src/nlp/features.py

```diff
@@ -95,6 +95,115 @@ def __call__(self):
         return self.pa_type
 
 
+@dataclass
+class MultiArray:
```

Comment:
Don't you think it could be simpler to just have `Array1D`, `Array2D`... up to a reasonable number ?
It would make things easier to read and maintain I guess.

Is there any reason to write it like this ?